libevm/EVMC.cpp

```diff
@@ -64,19 +68,28 @@ owning_bytes_ref EVMC::exec(u256& io_gas, ExtVMFace& _ext, const OnOpFunc& _onOp
     assert(_ext.depth <= static_cast<size_t>(std::numeric_limits<int32_t>::max()));
 
     auto gas = static_cast<int64_t>(io_gas);
-    EVM::Result r = execute(_ext, gas);
 
-    switch (r.status())
+    auto mode = toRevision(_ext.evmSchedule());
+    evmc_call_kind kind = _ext.isCreate ? EVMC_CREATE : EVMC_CALL;
+    uint32_t flags = _ext.staticCall ? EVMC_STATIC : 0;
+    assert(flags != EVMC_STATIC || kind == EVMC_CALL);  // STATIC implies a CALL.
+    evmc_message msg = {kind, flags, static_cast<int32_t>(_ext.depth), gas, toEvmC(_ext.myAddress),
+        toEvmC(_ext.caller), _ext.data.data(), _ext.data.size(), toEvmC(_ext.value),
+        toEvmC(0x0_cppui256)};
+    auto r = execute(_ext, mode, msg, _ext.code.data(), _ext.code.size());
+    auto output = bytesConstRef{r.output_data, r.output_size};
+
+    switch (r.status_code)
     {
     case EVMC_SUCCESS:
-        io_gas = r.gasLeft();
+        io_gas = r.gas_left;
         // FIXME: Copy the output for now, but copyless version possible.
-        return {r.output().toVector(), 0, r.output().size()};
+        return {output.toVector(), 0, output.size()};
```

Comment:
Why keep `bytesConstRef` here if this makes a copy anyway? Wouldn't it be simpler creating the vector here from `r.output_data` directly?